### PR TITLE
#3225 - Fixed issue with PackageGarbageCollector not identifying all …

### PR DESCRIPTION
**Pull request by davidjgonzalez**
…packages to clean up properly

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,8 +25,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com)
 
 - #3162 - Renovator MCP: ensure old source path is removed
 - #3205 - HttpClientFactory: Expose a method to customize the underlying HttpClient
-- #3209 - WARN org.apache.sling.models.impl.ModelAdapterFactory - Cannot provide default for java.util.List<java.lang.String>
-- #3197 - Encrypt user credentials in ACS Content Sync   
+- #3209 - WARN org.apache.sling.models.impl.ModelAdapterFactory - Cannot provide default for java.util.List<
+  java.lang.String>
```
Comment (davidjgonzalez) [inline CHANGELOG.md:29]:
```suggestion
- #3209 - WARN org.apache.sling.models.impl.ModelAdapterFactory - Cannot provide default for java.util.List<java.lang.String>
```

**Comment by davidjgonzalez** [comment]
@kwin not sure if you caught the last push - did some more testing and had to change things up -- basically

1. Dont use `createdDate` as this is (a) when the package was created -- so could be day/weeks/years before it was uploaded to AEM (b) its not on many packages in (atleast in AMS; maybe something with how AMS Cloud Manager works?)
2. Use `lastUnwrapped` as the new date when the package was uploaded/introduced to the AEM/JCR.
3. Continue using `lastUnpacked` as the date the package was installed to AEM/JCR.

File: CHANGELOG.md [hunk 2]
```diff
@@ -42,19 +46,22 @@ The format is based on [Keep a Changelog](http://keepachangelog.com)
 
 ## Removed
 
-- #3183 - Removed .wrap package including JackrabbitSessionIWrap and related classes which is no longer supported in Cloud Manager pipelines.
-  
+- #3183 - Removed .wrap package including JackrabbitSessionIWrap and related classes which is no longer supported in
```
Comment (kwin) [inline CHANGELOG.md:49]: I wouldn't break the line here either and in the occasions below

**Comment by kwin** [comment]
Probably https://github.com/davidjgonzalez/acs-aem-commons/blob/4010b059ccff691e54a6e20687597913bac3c1d0/bundle/src/main/java/com/adobe/acs/commons/packagegarbagecollector/PackageGarbageCollectionConfig.java#L33 should be updated as well, as package "age" is too fuzzy in this context. Rather refer to package upload date explicitly.

File: bundle/src/main/java/com/adobe/acs/commons/packagegarbagecollector/PackageGarbageCollectionConfig.java [hunk 3]
```diff
@@ -22,15 +22,15 @@
...
 
-    @AttributeDefinition(name = "Package Group Name", description = "The group name of the packages to remove")
+    @AttributeDefinition(name = "Package group name", description = "The group name of the packages to remove.")
     String groupName() default "";
 
-    @AttributeDefinition(name = "Max age of package", description = "Packages older than this (in days) will be removed. Default is 60 days")
+    @AttributeDefinition(name = "Max upload age of package", description = "Packages uploaded more than this age (in days) will be removed. Default is 60 days.")
     int maxAgeInDays() default 60;
 
     @AttributeDefinition(name = "Remove not installed packages", description = "Remove packages that are not installed (or have been uninstalled) and  the created date is older than maxAgeInDays. Default is false.")
```
Comment (kwin) [inline bundle/src/main/java/com/adobe/acs/commons/packagegarbagecollector/PackageGarbageCollectionConfig.java:36]: created date -> upload date

File: bundle/src/main/java/com/adobe/acs/commons/packagegarbagecollector/PackageGarbageCollectionConfig.java [hunk 4]
```diff
@@ -22,15 +22,15 @@
...
-    @AttributeDefinition(name = "Schedule", description = "Cron expression detailing when the garbage collection is run. Default runs at 02:30 every day")
+    @AttributeDefinition(name = "Schedule", description = "Cron expression detailing when the garbage collection is run. Default runs at 02:30 every day.")
     String scheduler() default "0 30 2 ? * * *";
 
-    @AttributeDefinition(name = "Package Group Name", description = "The group name of the packages to remove")
+    @AttributeDefinition(name = "Package group name", description = "The group name of the packages to remove.")
     String groupName() default "";
 
-    @AttributeDefinition(name = "Max age of package", description = "Packages older than this (in days) will be removed. Default is 60 days")
+    @AttributeDefinition(name = "Max upload age of package", description = "Packages uploaded more than this age (in days) will be removed. Default is 60 days.")
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/packagegarbagecollector/PackageGarbageCollectionConfig.java:33]:
```suggestion
    @AttributeDefinition(name = "Max upload age of package", description = "Packages uploaded more than the given amount of days ago will be removed. Default is 60 days.")
```

**Pull request merged**